### PR TITLE
Enhance ProjectService's adding of a user to project

### PR DESCRIPTION
**Pull request by lukas-slovak**
We added check for URI in roles from `userRoles` param to `ProjectService.addUserToProject`.
We edited method `ProjectService.getRoleByUri` to set URI to the returned role.

This solves the issue https://github.com/gooddata/gooddata-java/issues/766.

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: src/main/java/com/gooddata/project/ProjectService.java [hunk 1]
```diff
@@ -440,6 +438,8 @@ public User addUserToProject(final Project project, final Account account, final
         notNull(account, "account");
         notEmpty(account.getUri(), "account.uri");
         notNull(userRoles, "userRoles");
+        validateRoleURIs(userRoles);
+        noNullElements(Arrays.stream(userRoles).map(Role::getUri).collect(Collectors.toList()), "userRoles.uri");
```
Comment (liry) [inline src/main/java/com/gooddata/project/ProjectService.java]: redundant check

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by liry** [comment]
Please consider AT with use-case, that wasn't working (GET role and then POST to project).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/80/ : SUCCESS in 31s

**Pull request merged**